### PR TITLE
Expand test coverage

### PR DESCRIPTION
**Pull request by FrankvRaffe**
## Summary by CodeRabbit

* **New Features**
  * Allow initialization without crop or amendment inputs; system can now run spin-up simulations with soil and weather data alone.

* **Bug Fixes**
  * Fixed numeric type handling for amendment carbon content to prevent potential type issues.

* **Tests**
  * Expanded unit test coverage for initialization, input validation, and simulation functions to improve reliability and error detection.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+# rotsee UNRELEASED
+## changed
+* expanded unit tests for several functions
+* allow initialization if no crop or amendment inputs are provided 
+
```
[the remark this answers is not in the document]
Comment (FrankvRaffe) [reply NEWS.md:5]: @coderabbitai added in latest commit

File: tests/testthat/test-rc_input_rmf.R [hunk 2]
```diff
@@ -550,3 +550,25 @@ test_that("rc_input_rmf validates W_ET_REFACT at boundary values", {
...
+  weather <- create_weather()
+  
+  
+  # Run with no ET_REF_MONTH column and ET_ACT all NA
+  weather_noref <- copy(weather)[, W_ET_REF_MONTH := NULL]
+  expect_error(rc_input_rmf(dt = dt_crop, A_CLAY_MI = 20, dt.weather = weather_noref, dt.time = dt_time),
+               "W_ET_REF_MONTH is not available")
+  
+
+})
```
[the remark this answers is not in the document]
Comment (FrankvRaffe) [reply tests/testthat/test-rc_input_rmf.R:576]: @coderabbitai adapted in latest commit
